pyogrio/_io.pyx

```diff
@@ -1156,7 +1189,22 @@ def ogr_read(
 
         # Encoding is derived from the user, from the dataset capabilities / type,
         # or from the system locale
-        encoding = encoding or detect_encoding(ogr_dataset, ogr_layer)
+        if encoding:
+            if get_driver(ogr_dataset) == "ESRI Shapefile":
+                # Because SHAPE_ENCODING is set above, GDAL will automatically decode
+                # to UTF-8; ignore any encoding set by user
+                encoding = "UTF-8"
+
+        else:
+            if sql is not None and get_driver(ogr_dataset) == "ESRI Shapefile":
```

Comment:
Maybe mention here that this is fixed in GDAL and can be removed once GDAL3.9? becomes the lowest supported version?

Reply:
Added note that it may be fixed in 3.9.0 (hasn't been merged yet); I'm not setup to compile GDAL from source locally (don't have the time available to get everything ready for that), so I'm not able to definitely test the fix.